### DISCUSSION TITLE
Table Prints Hash Code Instead Of Table

### DISCUSSION THREAD

**Discussion by DevishaPadmaperuma** [Textualize/rich#1618] [Q&A] [answered]
Hi, I just started using Rich. When I call this function, it just prints `<rich.table.Table object at 0xXXXXX>` instead of a table.
The rows get added (placed a breakpoint and checked) but the table doesn't show anything.
```python3
def print_bill_items(self) -> None:

  table = Table(title="The Bill")

  table.add_column("Name", style="magenta")
  table.add_column("Price", style="cyan")
  table.add_column("Quantity", style="green")
  table.add_column("Total", style="red")

  for i in range(len(self.ar)):
      table.add_row(self.ar[i][0], str(self.ar[i][1]), str(self.ar[i][2]), str(self.ar[i][3]))
  print(table)
```
Also this is the function for print (override is a colour)
```python3
def print(prompt, override: str = None) -> None:
    if override is not None:
        console.print(f"[{override}]{prompt}[/{override}]")
    else:
        console.print(f"{prompt}")
```
I'm on Windows 10 and Python 3.9.6.
I'm sorry if I missed something obvious.

Thanks in advance!

**Comment by willmcgugan** [comment, marked as answer]
You're converting the table in to a string in the fstring, which will call `repr` on it. Just fo `console.print(prompt)`.

You won't be able to style the table by converting it in to a string. But you can set a style on the table, with `table.style = "red on white"` for example.

**Reply by DevishaPadmaperuma** [reply]
Thank you so much! I didn't think of that before. Sorry for wasting your time.
